src/Components/CreateImageWizard/validators.ts

```diff
@@ -58,6 +58,54 @@ export const isSnapshotValid = (dateString: string) => {
   return !isNaN(date.getTime()) && isSnapshotDateValid(date);
 };
 
+export const isUserNameValid = (userName: string) => {
+  const isLengthValid =
+    userName !== undefined && userName.length >= 2 && userName.length <= 32;
+  const isPatternValid = /^[a-zA-Z0-9_.][a-zA-Z0-9_.-]*[a-zA-Z0-9_.$-]?$/.test(
```

Comment:
I'm looking at the man pages for useradd and this is what it says:
>Usernames may contain only lower and upper case letters, digits, underscores, or dashes. They can end with a dollar sign. Dashes are not allowed at the beginning of the username. Fully numeric usernames and usernames . or .. are also disallowed. It is not recommended to use usernames beginning with . character as their home directories will be hidden in the ls output. Usernames may only be up to 32 characters long.

It didn't occur to me before to check how useradd works but I realized that's what gets called by the osbuild stage :facepalm: : https://github.com/osbuild/osbuild/blob/main/stages/org.osbuild.users

Let's just follow the man page rules exactly. So the minimum length should be 1. `..` and `.` would be special cases and forbidden.